### PR TITLE
Encrypt sec manager secret kms

### PR DESCRIPTION
**Pull request by chris-wilbur-wilson**
Checkov requires secret manager secrets to be encrypted with a CMK.
Functionality added to do this. The key is optional.

### CODE REVIEW

File: infra/terraform/modules/_auth/variables.tf [hunk 1]
```diff
@@ -22,3 +22,9 @@ variable "key_pair_rotation_period_days" {
   description = "The number of days between automatic scheduled rotations of the key pair"
   type        = number
 }
+
+variable "kms_key_arn" {
+  description = "kms key to encrypt secrets manager secret"
```
Comment (apgrucza) [inline infra/terraform/modules/_auth/variables.tf:27]:
```suggestion
  description = "The ARN of the KMS key used to encrypt the key pair"
```

File: infra/terraform/modules/_lambda/variables.tf [hunk 2]
```diff
@@ -30,3 +30,9 @@ variable "lambda_at_edge" {
   type        = bool
   default     = false
 }
+
+variable "kms_key_arn" {
+  description = "kms key to encrypt secrets manager secret"
```
Comment (apgrucza) [inline infra/terraform/modules/_lambda/variables.tf:35]:
```suggestion
  description = "The ARN of the KMS key used to encrypt the key pair"
```

File: infra/terraform/modules/okta_native/variables.tf [hunk 3]
```diff
@@ -58,3 +58,9 @@ variable "scope" {
   type        = string
   default     = "openid email"
 }
+
+variable "kms_key_arn" {
+  description = "kms key to encrypt secrets manager secret"
```
Comment (apgrucza) [inline infra/terraform/modules/okta_native/variables.tf:63]:
```suggestion
  description = "The ARN of the KMS key used to encrypt the key pair"
```

File: infra/terraform/modules/_auth/secrets.tf [hunk 4]
```diff
@@ -2,6 +2,7 @@ resource "aws_secretsmanager_secret" "key_pair" {
   name                    = "${var.name}/key-pair"
   recovery_window_in_days = 0
   tags                    = var.tags
+  kms_key_id              = var.kms_key_arn
```
Comment (apgrucza) [inline infra/terraform/modules/_auth/secrets.tf:5]: Does this work when `var.kms_key_arn` is `null`?

File: infra/terraform/modules/_lambda/iam.tf [hunk 5]
```diff
@@ -23,6 +27,14 @@ data "aws_iam_policy_document" "execution" {
 
     resources = ["arn:aws:logs:*:*:*"]
   }
+
+  statement {
+    actions   = ["kms:Decrypt","kms:GenerateDataKey"]
+    resources = [
+      "arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:key/*"
```
Comment (apgrucza) [inline infra/terraform/modules/_lambda/iam.tf]: You should be able to use `var.kms_key_arn` here instead

Comment (apgrucza) [reply infra/terraform/modules/_lambda/iam.tf]: I see, this is where you mentioned the need for a dynamic block, because the key might be null

**Comment by chris-wilbur-wilson** [comment]
Replaced with another PR minus the hundreds of commits!

**Pull request closed without merging**